src/unix/core.c

```diff
@@ -1102,3 +1102,66 @@ int uv_os_homedir(char* buffer, size_t* size) {
 
   return 0;
 }
+
+
+int uv_os_tmpdir(char* buffer, size_t* size) {
+  char* buf;
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  #define CHECK_ENV_VAR(name) \
+    buf = getenv(name);       \
+    if (buf != NULL)          \
+      goto return_buffer;
+
+  /* Check the TMPDIR, TMP, TEMP, and TEMPDIR environment variables in order */
+  CHECK_ENV_VAR("TMPDIR")
+  CHECK_ENV_VAR("TMP")
+  CHECK_ENV_VAR("TEMP")
+  CHECK_ENV_VAR("TEMPDIR")
+
+  #undef CHECK_ENV_VAR
+
+/*
+  buf = getenv("TMPDIR");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TMP");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TEMP");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TEMPDIR");
+
+  if (buf != NULL)
+    goto return_buffer;
+*/
+  /* No temp environment variables defined */
+  #if defined(__ANDROID__)
+    buf = "/data/local/tmp";
```

Comment:
`buf` should be `const char*` if you're pointing it to an immutable string.  (String literals are immutable unless assigned to a char array.)